.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,266 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+      - 'refs/pull/*'
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+
+env:
+  COMPOSER_ARGS_DEFAULT: --ansi --no-interaction --no-progress --prefer-dist
+  COMPOSER_ARGS_PHP8: --ansi --no-interaction --no-progress --prefer-dist --ignore-platform-req=php
+  CONFIG_FILE: .github/workflows/continuous-integration-config.json
+  PHP_DEFAULT_EXTENSIONS: '["mbstring","intl","readline"]'
+  PHP_DEFAULT_INI: '["memory_limit=-1"]'
+
+jobs:
+  lint-json:
+    name: Lint JSON files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: json-syntax-check
+        run: |
+          echo "Linting composer.json"
+          jq empty composer.json
+          for JSON in .github/workflows/*.json;do
+            echo "Linting $JSON"
+            jq empty $JSON
+          done
+
+  dependency-versions-matrix:
+    name: Dependencies to Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.dependency-versions-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: dependency-versions-matrix
+        run: |
+          DEPS=$(if [ -f composer.lock ]; then echo -n '["lowest", "locked", "latest"]'; else echo -n '["lowest", "latest"]'; fi)
+          echo "Testing against dependencies: ${DEPS}"
+          echo "::set-output name=check::${DEPS}"
+
+  config:
+    name: Retrieve configuration
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      json: ${{ steps.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: config
+        run: |
+          CONFIG=$(if [ -f ${CONFIG_FILE} ]; then jq -c '.' < ${CONFIG_FILE} ; else echo -n "{}" ; fi)
+          echo "Using config: $CONFIG"
+          echo "::set-output name=json::${CONFIG}"
```

Comment:
Confused about this: the file is at a fixed location, so why do we need a whole job for it?

Reply:
This does a few things.

First, if the config file is _not_ present, it provides some default contents (an empty object).

Second, `jq -c '.'` ensures we have a compact JSON string suitable for use with GHA expressions.

Third, this puts the JSON into an output, which means we can then reference via GHA context, which is what allows us to use it _in_ a GHA expression in the first place.

It's a roundabout way of getting the config out of a file and into the GHA context, essentially.